ncrypt/crypt_gpgme.c

```diff
@@ -3465,12 +3464,12 @@ static const char *crypt_format_str(char *buf, size_t buflen, size_t col, int co
         if (key->kobj->subkeys && (key->kobj->subkeys->timestamp > 0))
           tt = key->kobj->subkeys->timestamp;
 
-        tm = localtime(&tt);
+        tm = mutt_date_localtime(tt);
```

Comment:
I think there's a bug, here.
The original code:
- set `tt = 0`
- follow path where the `if` fails
- `localtime(0)`
- Result '1970-01-01'

The new code:
- set `tt = 0`
- follow path where the `if` fails
- `mutt_date_localtime(0)`
- Result '2019-03-13'

A simple fix would be to change the constant `MUTT_DATE_NOW` to something distinct and unlikely.
`#define MUTT_DATE_NOW -9999`

Reply:
That's great catch, thanks!

After taking a look I realised tt is only used there in that function, could we refactor it even more by removing `tt` var?

```
    {
-     time_t tt = 0;
-
      if (key->kobj->subkeys && (key->kobj->subkeys->timestamp > 0))
-       tt = key->kobj->subkeys->timestamp;
+       tm = mutt_date_localtime(key->kobj->subkeys->timestamp);
+     else
+       tm = mutt_date_localtime(0);
-
-     tm = mutt_date_localtime(tt);
    }

Reply:
how about

```c
tm = mutt_date_localtime(
    (key->kobj->subkeys && (key->kobj->subkeys->timestamp > 0))
    ? key->kobj->subkeys->timestamp  : 0)
```

:)

Reply:
@ghikio 
Yes that looks tidier (but still legible).
Perhaps add a comment:
```c
tm = mutt_date_localtime(0); // Default to 1970-01-01
```

Reply:
@gahr Inline ifs scare me everytime :)